### PR TITLE
Remove pylab to eliminate import conflicts that crash the Jupyter kernel

### PR DESCRIPTION
**Pull request by zacsimile**
Together, these imports crash the Jupyter notebook kernel:

```
import matplotlib
matplotlib.use('wxagg')
import pylab
```

If we comment out `matplotlib.use('wxagg')` or `import pylab`, everything runs fine. To fix this, I've switched our only use of `pylab` in `VisGUI.py` to an equivalent `matplotlib` call.

### CODE REVIEW

**Comment by David-Baddeley** [comment]
This has been on the TODO list for ages - thanks! Some of this might go a little beyond what was strictly required (I wouldn't have bothered with any of the stuff in 'Loft', and likewise would have probably left the stuff under contrib alone).

I've got 3 possible concerns:

- substituting `pylab.fft*` with `scipy.fftpack.*` - I'm not sure where things are at the moment, but there was a time where these were not strictly interchangeable (even using different libraries under the hood). I'm not sure if that is still the case (at one point `scipy.fftpack` was missing `fftshift` and `ifftshift` and then the behaviour of the scipy version was incorrect. We would need to do some checking here.
- we do some nasty monkey patching of `pylab.cm` in `extraCMaps` - we need to double check that this maps 1:1 to `matplotlib.cm`
- I don't think your replacement is 1:1 for the `from pylab import *` in the interactive shell module. I also think the pylab import route is probably cleaner here. 

Looking back to your original issue, are you sure that this can't be resolved by a `%gui wx` magic at the start of your notebook? If this is the case, I'd prefer to keep the pylab imports for the shell.

**Comment by zacsimile** [comment]
Hi David!

About halfway through, I put the import pylab back in the shell because I realized it wasn't 1:1. I did double check that `scipy.fftpack` now has `fftshift` and `ifftshift`. I also double-checked that `pylab.cm` is the same as `matplotlib.cm`. The only thing I'm not 100% on is if `__getattribute__` works for `matplotlib.cm`.

I have a `% gui wx` call. Even after eliminating `pylab`, it still crashes. It looks like it has to do with the order we call `matplotlib.use('wxagg')`. I haven't solved it yet.

**Comment by zacsimile** [comment]
At some point it got easier to just edit every single file using `pylab`. Sorry if I hit some stuff you'd prefer to be left alone. I did do roughly one commit per file, and I included the file name in the commit, so it should be fairly easy to cherry-pick if you like.

File: PYME/Acquire/Hardware/driftTracking.py [hunk 1]
```diff
@@ -6,7 +6,8 @@
 """
 
 import numpy as np
-from pylab import fftn, ifftn, fftshift, ifftshift
+# from pylab import fftn, ifftn, fftshift, ifftshift
+from scipy.fftpack import fftn, ifftn, fftshift, ifftshift
```
Comment (David-Baddeley) [inline PYME/Acquire/Hardware/driftTracking.py]:
```suggestion
from numpy.fft import fftn, ifftn, fftshift, ifftshift

```

File: PYME/Analysis/PSFGen/fourier.py [hunk 2]
```diff
@@ -22,7 +22,8 @@
 """generate a phase ramp psf unp.sing fourier optics"""
 import matplotlib.pyplot as plt
 import numpy as np
-from pylab import ifftshift, ifftn, fftn, fftshift
+# from pylab import ifftshift, ifftn, fftn, fftshift
+from scipy.fftpack import ifftshift, ifftn, fftn, fftshift
```
Comment (David-Baddeley) [inline PYME/Analysis/PSFGen/fourier.py]:
```suggestion
from numpy.fft import ifftshift, ifftn, fftn, fftshift
```

File: PYME/Analysis/PSFGen/fourierHNA2.py [hunk 3]
```diff
@@ -23,7 +23,8 @@
 #from pylab import *
 import matplotlib.pyplot as plt
 import numpy as np
-from pylab import ifftshift, ifftn, fftn, fftshift
+# from pylab import ifftshift, ifftn, fftn, fftshift
+from scipy.fftpack import ifftshift, ifftn, fftn, fftshift
```
Comment (David-Baddeley) [inline PYME/Analysis/PSFGen/fourierHNA2.py]:
```suggestion
from numpy.fft import ifftshift, ifftn, fftn, fftshift
```

File: PYME/Analysis/deTile.py [hunk 4]
```diff
@@ -25,7 +25,8 @@
 from scipy import fftpack, ndimage
 from PYME.Acquire.Hardware import splitter
 
-from pylab import ifftshift, ifftn, fftn, fftshift
+# from pylab import ifftshift, ifftn, fftn, fftshift
+from scipy.fftpack import ifftshift, ifftn, fftn, fftshift
```
Comment (David-Baddeley) [inline PYME/Analysis/deTile.py]:
```suggestion
from numpy.fft import ifftshift, ifftn, fftn, fftshift
```

File: PYME/DSView/modules/composite.py [hunk 5]
```diff
@@ -128,8 +128,8 @@ def OnApply(self, event):
         self.dsviewer.do.OnChange()
         
     def OnCorrelate(self, event):
-        from scipy.fftpack import fftn, ifftn
-        from pylab import fftshift, ifftshift
+        from scipy.fftpack import fftn, ifftn, fftshift, ifftshift
```
Comment (David-Baddeley) [inline PYME/DSView/modules/composite.py]:
```suggestion
        from numpy.fft import fftn, ifftn, fftshift, ifftshift
```

File: PYME/experimental/SimReconstruction.py [hunk 6]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 import numpy as np
-from pylab import fftn, ifftn, fftshift, ifftshift
-
+# from pylab import fftn, ifftn, fftshift, ifftshift
+from scipy.fftpack import fftn, ifftn, fftshift, ifftshift
```
Comment (David-Baddeley) [inline PYME/experimental/SimReconstruction.py]:
```suggestion
from numpy.fft import fftn, ifftn, fftshift, ifftshift
```

File: PYME/misc/shellutils.py [hunk 7]
```diff
@@ -308,7 +308,9 @@ def binSum(binVar, indepVar, bins):
 def frc(image):
     from PYME.Analysis import binAvg        
     import numpy as np
-    import pylab
+    # import pylab
+    import matplotlib.pyplot as plt
+    from scipy.fftpack import fft2, fftshift
```
Comment (David-Baddeley) [inline PYME/misc/shellutils.py]:
```suggestion
    from numpy.fft import fft2, fftshift
```

File: PYME/misc/shellutils.py [hunk 8]
```diff
@@ -354,15 +356,15 @@ def frc(image):
...
 
-    pylab.show()
+    plt.show()
 
     return H1, H2, R, bmr/np.sqrt(bm1*bm2), bn, bm, bm1, bm2, rB
 
 def abscorrel(a,b):
-    from scipy.fftpack import fftn, ifftn
-    from pylab import fftshift, ifftshift
+    from scipy.fftpack import fftn, ifftn, fftshift, ifftshift
```
Comment (David-Baddeley) [inline PYME/misc/shellutils.py]:
```suggestion
    from numpy.fft import fftn, ifftn, fftshift, ifftshift
```

**Pull request merged**